partiql-lang/src/main/kotlin/org/partiql/lang/syntax/impl/PartiQLPigVisitor.kt

```diff
@@ -1325,16 +1325,48 @@ internal class PartiQLPigVisitor(
         canLosslessCast(expr, type, metas)
     }
 
-    override fun visitFunctionCall(ctx: PartiQLParser.FunctionCallContext) = PartiqlAst.build {
+    override fun visitFunctionCall(ctx: PartiQLParser.FunctionCallContext): PartiqlAst.Expr = PartiqlAst.build {
         val nameCtx = ctx.qualifiedName()
         val name = if (nameCtx.qualifier.isNotEmpty()) {
             error("Legacy AST does not support qualified function names")
         } else {
             nameCtx.name.getString().lowercase()
         }
-        val args = ctx.expr().map { visitExpr(it) }
-        val metas = ctx.start.getSourceMetaContainer()
-        call(name, args = args, metas = metas)
+        // COUNT(*) turns into COUNT(1) in legacy PIG AST
+        if (ctx.ASTERISK() != null) {
+            if (name == "count") {
+                return@build callAgg(
+                    all(),
+                    name,
+                    lit(ionInt(1)),
+                    nameCtx.name.getSourceMetaContainer() + metaContainerOf(IsCountStarMeta.instance)
+                )
+            } else {
+                // Do not permit wildcard for other functions; matches existing behavior
+                error("Given wildcard as argument for non-COUNT function")
+            }
+        }
+        if (!name.isAggregateCall()) {
+            // Scalar fn call
+            val args = ctx.expr().map { visitExpr(it) }
+            val metas = ctx.start.getSourceMetaContainer()
+            return@build call(name, args = args, metas = metas)
+        }
+        // Aggregate fn call
+        val strategy = getStrategy(ctx.setQuantifierStrategy(), default = all())
+        val args = ctx.expr()
+        if (args.size != 1) {
+            error("expect only one argument to aggregate function call")
+        }
+        val arg = visitExpr(args.first())
+        val metas = nameCtx.name.getSourceMetaContainer()
+        return@build callAgg(strategy, name, arg, metas)
+    }
+
+    private fun String.isAggregateCall(): Boolean {
```

Comment:
(self-review) in the legacy PIG AST and conversion from `partiql-ast` to PIG AST, keep the same behavior as before with respect to supported aggregations. E.g.
- `SUM(*)` and `F(*)` will still error
- Aggregations require exactly `1` argument

Hard-coding the aggregations in the conversion to PIG AST since this is legacy code that will be removed by the v1 release.